From df3573982b62886e5bda3fe1e393b453019e0081 Mon Sep 17 00:00:00 2001
From: Gaurav Mishra <mishra.gaurav@siemens.com>
Date: Thu, 23 Dec 2021 15:01:11 +0530
Subject: [PATCH] feat(nomos): See file regex to include view

While looking out for files with `See-file.LICENSE` regex, look for
"view" as an alternative for "see".

For example, "please view the LICENSE file" and "please see the LICENSE
file" to be matched.

Signed-off-by: Gaurav Mishra <mishra.gaurav@siemens.com>

Upstream-Status: Backport [https://github.com/fossology/fossology/commit/df3573982b62886e5bda3fe1e393b453019e0081]
---

diff --git a/src/nomos/agent/STRINGS.in b/src/nomos/agent/STRINGS.in
index 320ad725a..ee9e2816b 100755
--- a/src/nomos/agent/STRINGS.in
+++ b/src/nomos/agent/STRINGS.in
@@ -6463,7 +6463,7 @@ k
 #
 %ENTRY% _LT_SEE_LICENSE_2
 %KEY% "(condit|vers)ion"
-%STR% "under (certain|the) conditions (refer t(o|o the)|for details (read|read the|refer t(o|o the)|see|see the)|listed in the) (file =FEW= \<licen[cs]e\>|\<licen[cs]e\> file)"
+%STR% "under (certain|the) conditions (refer t(o|o the)|for details (read|read the|refer t(o|o the)|see|view|see the)|listed in the) (file =FEW= \<licen[cs]e\>|\<licen[cs]e\> file)"
 #
 %ENTRY% _LT_SEE_LICENSE_3
 %KEY% "licen[cs]"
@@ -6492,23 +6492,23 @@ k
 #
 %ENTRY% _LT_SEE_LICENSE_9
 %KEY% "licen[cs]"
-%STR% "see license?(qla|qlcnic|qlge) =SOME= for copyright and licensing details"
+%STR% "(see|view) license?(qla|qlcnic|qlge) =SOME= for copyright and licensing details"
 #
 %ENTRY% _LT_SEE_LICENSE_10
-%KEY% "licen[cs]" 
-%STR% "licen[cs]e see licen[cs]e?txt" 
+%KEY% "licen[cs]"
+%STR% "licen[cs]e (see|view) licen[cs]e?txt"
 #
 %ENTRY% _LT_SEE_LICENSE_11
 %KEY% "licen[cs]"
-%STR% "see the accompanying file license =SOME= for terms of use"
+%STR% "(see|view) the accompanying file license =SOME= for terms of use"
 #
 %ENTRY% _LT_SEE_LICENSE_12
 %KEY% "licen[cs]"
-%STR% "license please see accompanying LICENSE"
+%STR% "license please (see|view) accompanying LICENSE"
 #
 %ENTRY% _LT_SEE_LICENSE_13
 %KEY% "licen[cs]"
-%STR% "see =FEW= license file =FEW= for more information"
+%STR% "(see|view) =FEW= license file =FEW= for more information"
 #
 %ENTRY% _LT_SEE_LICENSE_14
 %KEY% "licen[cs]"
@@ -6528,7 +6528,7 @@ k
 #
 %ENTRY% _LT_SEE_LICENSE_18
 %KEY% "licen[cs]"
-%STR% "see (the )?license( file|\.?txt) =FEW= for licens(ing terms|e information)"
+%STR% "(see|view) (the )?license( file|\.?txt) =FEW= for licens(ing terms|e information)"
 #
 %ENTRY% _LT_SEE_LICENSE_19
 %KEY% "licen[cs]"
@@ -6540,13 +6540,18 @@ k
 #####
 # These MUST be the last 2 checks for license references in the file "license"
 #####
+# Following two strings have (see|view) as alternative which does not work with
+# CHECKSTR. Thus the NOCHECK
+#####
 %ENTRY% _LT_SEE_LICENSE_FINAL1
-%KEY% "\<(see|sell|sale|sold|charge)\>"
-%STR% "(distribut|copy|mod) =SOME= for details see th(e|is) (file =FEW= \<licen[cs]e\>|\<licen[cs]e\> file)"
+%NOCHECK%
+%KEY% "\<(see|sell|sale|sold|charge|view)\>"
+%STR% "(distribut|copy|mod) =SOME= for details (see|view) th(e|is) (file =FEW= \<licen[cs]e\>|\<licen[cs]e\> file)"
 #
 %ENTRY% _LT_SEE_LICENSE_FINAL2
-%KEY% "\<(see|sell|sale|sold|charge)\>"
-%STR% "see the (file =FEW= \<licen[cs]e\>|\<licen[cs]e\> file) =SOME= (distribut|dir|copy|us(e|age)|mod|free soft)"
+%NOCHECK%
+%KEY% "\<(see|sell|sale|sold|charge|view)\>"
+%STR% "(see|view) the (file =FEW= \<licen[cs]e\>|\<licen[cs]e\> file) =SOME= (distribut|dir|copy|us(e|age)|mod|free soft)"
 #
 %ENTRY% _LT_SEE_LICENSE_FINAL3
 %NOCHECK%
-- 
2.39.2

